packages/entrykit/src/onboarding/deposit/useChainBalances.ts

```diff
@@ -1,22 +1,41 @@
 import { useQuery } from "@tanstack/react-query";
-import { Hex } from "viem";
-import { useChains, useConfig as useWagmiConfig } from "wagmi";
+import { Address, Chain } from "viem";
+import { useAccount, useConfig as useWagmiConfig } from "wagmi";
 import { getBalance } from "wagmi/actions";
 
 export type UseChainBalancesOptions = {
-  address: Hex;
+  chains: Chain[];
 };
 
-export function useChainBalances({ address }: UseChainBalancesOptions) {
+export function useChainBalances({ chains }: UseChainBalancesOptions) {
+  const userAccount = useAccount();
   const wagmiConfig = useWagmiConfig();
-  const chains = useChains();
   const chainIds = chains.map((chain) => chain.id);
   return useQuery({
     queryKey: ["chainBalances", chainIds],
     queryFn: async () => {
-      return await Promise.all(chains.map(async (chain) => getBalance(wagmiConfig, { chainId: chain.id, address })));
+      const balances = await Promise.all(
+        chains.map(async (chain) => {
+          try {
+            return await getBalance(wagmiConfig, { chainId: chain.id, address: userAccount.address as Address });
+          } catch (error) {
+            console.warn(`Failed to fetch balance for chain ${chain.id}:`, error);
+            return undefined;
+          }
+        }),
+      );
```

Comment:
you can do this without mutating the return type of the promise

```ts
const balances = Promise.allSettled(chains.map((chain) => getBalance(wagmiConfig, { chainId: chain.id, address })))
```

Reply:
Neat, didn't think about this, updated

Reply:
returning an array of `chain` + `balance` means you also don't need to do the look up